src/database.ts

```diff
@@ -478,8 +470,13 @@ class Database extends common.GrpcServiceObject {
     this._observabilityOptions = instance._observabilityOptions;
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     this.requestStream = instance.requestStream as any;
-    this.pool_.on('error', this.emit.bind(this, 'error'));
-    this.pool_.open();
+    this.sessionFactory_ = new SessionFactory(this, name, poolOptions);
+    this.pool_ = this.sessionFactory_.getPool();
```

Comment:
Why do we need pool_ object here. This should be replaced with `sessionFactory_` everywhere right ?

Reply:
some RW methods like `runTransaction`, `batchWriteAtLeastOnce` are getting the session from the Session Pool. Since, the changes to these RW methods is out of the scope in the first phase of the development, hence we will need to ensure the pool_ object for their use